packages/rocketchat-federation/server/peerDNS.js

```diff
@@ -33,13 +33,13 @@ class PeerDNS {
 	//
 	// ########
 	register(peerConfig) {
-		const { uniqueId, domain, url, public_key } = peerConfig;
+		const { uniqueId, domain, url, public_key, cloud_token } = peerConfig;
 
 		this.log(`Registering peer with domain ${ domain }...`);
 
 		// Attempt to register peer
 		try {
-			peerHTTP.request(this.HubPeer, 'POST', '/api/v1/peers', { uniqueId, domain, url, public_key }, { total: 5, stepSize: 1000, tryToUpdateDNS: false });
+			peerHTTP.request(this.HubPeer, 'POST', '/api/v1/peers', { uniqueId, domain, url, public_key, cloud_token }, { total: 5, stepSize: 1000, tryToUpdateDNS: false });
```

Comment:
Out of curiosity why in the payload?  Could just pass in the header as bearer token if exists.  Then hub side if a bearer token is passed we would handle it.  

This would just make more consistent with how its used every where else with the cloud token